### PR TITLE
Publish identities to SSE state

### PR DESCRIPTION
**Pull request by tadaskay**
Ref #1844

Identity list is fetched on the initial state construction and then updated on:

- AppTopicIdentityCreated (re-fetches the whole list, to ensure the same order as the identity manager)
- AppTopicIdentityRegistration
- AppTopicBalanceChanged (Which is also triggered by AppTopicIdentityUnlock)

### CODE REVIEW

File: core/state/event/event.go [hunk 1]
```diff
@@ -29,10 +30,18 @@ const AppTopicState = "State change"
...
+	NATStatus  NATStatus        `json:"natStatus"`
+	Services   []ServiceInfo    `json:"serviceInfo"`
+	Sessions   []ServiceSession `json:"sessions"`
+	Consumer   ConsumerState    `json:"consumer"`
+	Identities []Identity       `json:"identities"`
+}
+
+// Identity represents identity and its status.
+type Identity struct {
+	Address            string                      `json:"id"`
```
Comment (Waldz) [inline core/state/event/event.go]: These JSON mapping not needed here, as you're remapping to `contract.IdentityDTO` later.
Is this intermediate structure needed at all?

File: tequilapi/endpoints/sse_handler.go [hunk 2]
```diff
@@ -222,6 +224,15 @@ type consumerConnectionRes struct {
 }
 
 func mapState(event stateEvent.State) stateRes {
+	identitiesRes := make([]contract.IdentityDTO, len(event.Identities))
+	for idx, identity := range event.Identities {
+		identityRes := contract.IdentityDTO{
```
Comment (Waldz) [inline tequilapi/endpoints/sse_handler.go]: confusing place - `contract` package with single model

Comment (tadaskay) [reply tequilapi/endpoints/sse_handler.go]: This is different PR: https://github.com/mysteriumnetwork/node/pull/1906
I've extracted the contract to a separate package for re-use between the server and the client. Not refactoring all at once, but rather following the boy scout rule. What is confusing?

File: core/state/state.go [hunk 3]
```diff
@@ -100,31 +112,45 @@ func NewKeeper(
...
 }
 
+func (k *Keeper) fetchIdentities() []stateEvent.Identity {
+	providedIdentities := k.deps.IdentityProvider.GetIdentities()
+	identities := make([]event.Identity, len(providedIdentities))
+	for idx, providedIdentity := range providedIdentities {
+		status, err := k.deps.IdentityRegistry.GetRegistrationStatus(providedIdentity)
+		if err != nil {
+			log.Warn().Err(err).Msgf("Could not get registration status for %s", providedIdentity.Address)
+			status = registry.Unregistered
```
[the remark this answers is not in the document]
Comment (tadaskay) [reply core/state/state.go]: If an error is returned, the result should generally be discarded. What you are saying is an implementation detail of the registry.

File: core/state/event/event.go (hunk 1 above)
Comment (tadaskay) [reply core/state/event/event.go]: What are you suggesting to replace it with?

**Pull request merged**